modules/tinymce/src/core/main/ts/selection/RangeNodes.ts

```diff
@@ -17,18 +21,27 @@ const getSelectedNode = (range: Range): Node => {
 };
 
 const getNode = (container: Node, offset: number): Node => {
-  if (container.nodeType === 1 && container.hasChildNodes()) {
-    if (offset >= container.childNodes.length) {
-      offset = container.childNodes.length - 1;
-    }
-
-    container = container.childNodes[offset];
+  if (NodeType.isElement(container) && container.hasChildNodes()) {
+    const childNodes = container.childNodes;
+    const safeOffset = Num.clamp(offset, 0, childNodes.length - 1);
+    return childNodes[safeOffset];
+  } else {
+    return container;
   }
+};
 
-  return container;
+/** @deprecated Use getNode instead */
```

Comment:
We're deprecating this but also using it in CaretPosition above?

Reply:
Yes, it's because `CaretPosition` has always needed this logic and we can't change it without breaking a heap of things. I merged the safer option from `RangeWalk` which clamps the offset, which is now the replacement `getNode` since it's safer. So this just makes sure we don't attempt to use the unsafe version in future.